lib/shared/puppet_forge/module_name.rb

```diff
@@ -0,0 +1,23 @@
+module PuppetForge
+  # Normalize Forge module names
+  #
+  # Forge modules use '-' as the separator between the username and modulename
+  # in the module slug, but the Puppet Forge uses '/' as the separator on module
+  # pages and in the URL. This class hides the distinction by always using '-' as
+  # the separator for normal use. If the name needs to use the slash separated name
+  # then the #forge_name method can be used.
+  #
+  # @api private
+  class ModuleName < String
+
+    def initialize(*args)
+      super
+      tr!('/', '-')
+    end
+
+    # @return [String] The slash separated module name.
+    def forge_name
+      String.new(self).tr('-', '/')
+    end
```

Comment:
Ah yes, I agree that returning a string is preferable there.

As for the larger question of whether the class should exist, it seemed reasonable to me but this article leads me to believe there are a lot of dragons to watch out for when subclassing Ruby core: http://words.steveklabnik.com/beware-subclassing-ruby-core-classes

Specifically there are circumstances where Ruby will not call the initializer of your custom String subclass which seems like it might defeat the whole purpose of this class. :/ Something like his approach using Forwardable might be preferable so that we can explicitly control what gets delegated to String.

I think this is OK to merge but we probably should revisit at some point.